engine/src/test/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionTest.java

```diff
@@ -232,6 +233,33 @@ public void shouldWriteEventsForDeletedProcessWithoutRunningInstances() {
     verifyResourceIsDeleted(processDefinitionKey);
   }
 
+  @Test
+  public void shouldWriteEventsForDeletedProcessWithBannedInstances() {
+    // given
+    final var processId = helper.getBpmnProcessId();
+    final var processDefinitionKey =
+        engine
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId).startEvent().userTask().endEvent().done())
+            .deploy()
+            .getValue()
+            .getProcessesMetadata()
+            .get(0)
+            .getProcessDefinitionKey();
+    final long processInstanceKey = engine.processInstance().ofBpmnProcessId(processId).create();
+    final var bannedInstanceState =
+        (MutableBannedInstanceState) engine.getProcessingState().getBannedInstanceState();
+    bannedInstanceState.banProcessInstance(processInstanceKey);
```

Comment:
ℹ️ This is slightly dirty, but it's the easiest way to mark my instance as banned. There is no simple and reliable way to cause an instance to banned naturally. Besides that, if there was this test would start failing once we fixed that bug.

Reply:
I think this is a nice and reliable approach to test this. I agree that simulating a PI ban would produce a more flaky test.